examples/audio_classifier/app/src/main/java/com/google/mediapipe/examples/audioclassifier/AudioClassifierHelper.kt

```diff
@@ -0,0 +1,253 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.mediapipe.examples.audioclassifier
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.media.AudioFormat
+import android.media.AudioRecord
+import android.media.MediaRecorder
+import android.os.SystemClock
+import android.util.Log
+import com.google.mediapipe.tasks.audio.audioclassifier.AudioClassifier
+import com.google.mediapipe.tasks.audio.audioclassifier.AudioClassifierResult
+import com.google.mediapipe.tasks.audio.core.RunningMode
+import com.google.mediapipe.tasks.components.containers.AudioData
+import com.google.mediapipe.tasks.components.containers.AudioData.AudioDataFormat
+import com.google.mediapipe.tasks.components.containers.Classifications
+import com.google.mediapipe.tasks.core.BaseOptions
+import com.google.mediapipe.tasks.core.Delegate
+import java.util.concurrent.ScheduledThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+
+class AudioClassifierHelper(
+    val context: Context,
+    var currentModel: Int = YAMNET_MODEL,
+    var classificationThreshold: Float = DISPLAY_THRESHOLD,
+    var overlap: Int = DEFAULT_OVERLAP,
+    var numOfResults: Int = DEFAULT_NUM_OF_RESULTS,
+    var currentDelegate: Int = DELEGATE_CPU,
+    var runningMode: RunningMode = RunningMode.AUDIO_CLIPS,
+    var listener: ClassifierListener? = null,
+) {
+
+    private var recorder: AudioRecord? = null
+    private var executor: ScheduledThreadPoolExecutor? = null
+    private var audioClassifier: AudioClassifier? = null
+    private val classifyRunnable = Runnable {
+        recorder?.let { classifyAudioAsync(it) }
+    }
+
+    init {
+        initClassifier()
+    }
+
+    @SuppressLint("MissingPermission")
+    fun initClassifier() {
+        // Set general detection options, e.g. number of used threads
+        val baseOptionsBuilder = BaseOptions.builder()
+
+        // Use the specified hardware for running the model. Default to CPU.
+        // Possible to also use a GPU delegate, but this requires that the classifier be created
+        // on the same thread that is using the classifier, which is outside of the scope of this
+        // sample's design.
+        when (currentDelegate) {
+            DELEGATE_CPU -> {
+                // Default
+                baseOptionsBuilder.setDelegate(Delegate.CPU)
+            }
+            DELEGATE_GPU -> {
+                baseOptionsBuilder.setDelegate(Delegate.GPU)
+            }
+        }
+
+        when (currentModel) {
+            YAMNET_MODEL -> {
```

Comment:
Since we're only using one model and not showing a radio button for changing it in our UI, let's remove this when statement and related constants. We can readd them later if we support an additional model.